benchmarks/video-analytics/object_recognition/recog.py

```diff
@@ -125,7 +126,7 @@ def Recognise(self, request, context):
         if self.transferType == S3:
             log.info("retrieving target frame '%s' from s3" % request.s3key)
             with tracing.Span("Frame fetch"):
-                frame = storage.get(request.s3key)
+                frame = pickle.loads(storageBackend.get(request.s3key))
```

Comment:
There exist cases where pickling is not needed (such as [when the buckets contain textual data](https://github.com/vhive-serverless/vSwarm/blob/aws_lambda_support/benchmarks/map-reduce/mapper/mapper.py#L48)).
Since the decision to pickle or not depends on the workload, I believe it is best to handle it there along with all other preprocessing and postprocessing of data.

Reply:
Okay, makes sense.